### DISCUSSION TITLE
Examples or libraries for opening, manipulating and saving a .nupkg file.

### DISCUSSION THREAD

**Discussion by The-Futurist** [NuGet/Home#11285] [Q&A]
Are there any code resources I could refer to if I wanted to write code that can open a `nupkg `file make some changes and save the file?

I'm considering writing code that can add an XML node to the `nusepc `in cases where its impossible to rebuild the package from its input files.

e.g.

![image](https://user-images.githubusercontent.com/12262952/135544451-c23f6386-1557-4960-8435-15a66ac3d186.png)

packages that were initially created without that `<repository>` node need to have it added to them but we cannot create the package afresh from the original inputs and instead must modify the data.

This is forced upon us because we need to push a hundred+ packages of ours (these are all private packages and used purely internally) into Github but they insist that that attribute (or `<RepositoryUrl>`) is present in order to accept the push request.

Recreating the packages is not possible in many cases because they were built many years ago and the build machines are gone and it is unlikely we could accurately recreate the packages (for example many refer to ,Net code that was in the GAC and is no longer etc etc).

Modifying the packages this way is probably the least risk way of doing this.

Thanks.

**Comment by erdembayar** [comment]
Hi.
Basically it's just normal zip file, you can rename nupkg file to zip extension (then extract) and make changes (then pack again) and rename back to nupkg. 
I believe it's doable with powershell scripts. You may manipulate xml with powershell, or you may write simple console app iterate through your nupkg files and do above. But it's not recommended for public packages.
